### PR TITLE
[Backport stable/8.1] feat: run message TTL checker concurrently to processing

### PR DESCRIPTION
**Pull request by korthout**
## Description

Backports #11792 to `stable/8.1`.

Conflicts had to be resolved in many files, particularly in 834e5d415429740899c2347b02aa52023bc039ef. Please consider reviewing this with extra care.

## Related issues

relates to https://github.com/camunda/zeebe/issues/11761

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnVariableMappingBehavior.java [hunk 1]
```diff
@@ -141,11 +141,18 @@ private long getVariableScopeKey(final BpmnElementContext context) {
...
     }
   }
+
+  private boolean isErrorEvent(final ExecutableFlowNode element) {
+    if (element instanceof final ExecutableCatchEventElement catchEvent) {
+      return catchEvent.isError();
+    } else {
+      return false;
+    }
+  }
```
Comment (korthout) [inline engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnVariableMappingBehavior.java:157]: 👎 Oops, I accidentally added this. I'll remove it

**Review by lenaschoenburg — APPROVED** [review]
:rocket: Thanks for backporting @korthout, the conflict resolution must have been fun :grimacing:

**Comment by korthout** [comment]
> the conflict resolution must have been fun 😬

@oleschoenburg It was less bad then it looks. Also, I had [`git mergetool` with IntelliJ-IDEA](https://dev.to/korthout/git-mergetool-for-java-files-26jd) at my side 🚀

**Comment by korthout** [comment]
bors merge

**Comment by korthout** [comment]
> Build failed:
> 
> * [Test summary](https://github.com/camunda/zeebe/actions/runs/4341609793/jobs/7581704719)

Seems to have been a fluke.

bors merge

**Comment by korthout** [comment]
bors merge

**Pull request merged**